src/components/MaterialUI/ScopeSelector/TreeView.js

```diff
@@ -3,49 +3,70 @@ import TreeViewMui from "@material-ui/lab/TreeView";
 import TreeItem from "./TreeItem";
 import { scopeTypes } from "./../../../constants";
 
-const TreeView = ({ rootId, getScope, selectedScope, closeSelector }) => {
-	const [expanded, setExpanded] = React.useState(selectedScope.scopePath);
-	const [selected, setSelected] = React.useState(selectedScope.id);
-
-	const virtualScopes = [];
+export const BaseTreeView = ({
+	rootId,
+	getScope,
+	closeSelector,
+	initExpanded,
+	selected,
+	setSelected,
+	redirectOnSelect,
+}) => {
+	const [expanded, setExpanded] = React.useState(initExpanded);
 
 	const handleToggle = (_, nodeIds) => {
 		setExpanded(nodeIds);
 	};
 
-	const handleSelect = (_, nodeId) => {
-		if (virtualScopes.includes(nodeId) === false) {
-			setSelected(nodeId);
-		}
+	const onScopeSelect = (_, nodeId) => {
+		const newSelected = !Array.isArray(selected)
+			? nodeId
+			: selected.includes(nodeId)
+			? selected.filter(id => id !== nodeId)
+			: [nodeId, ...selected];
+
+		setSelected(newSelected);
+		closeSelector(_, newSelected);
```

Comment:
Usually parameter can/should be placeholder when we are not going to use it in handler. However the fact that now we are passing it to close selector means that we are started to use it in some way. So, now we need regular parameter (like event) instead of placeholder.